coordinator/src/trade/mod.rs

```diff
@@ -513,7 +568,205 @@ impl TradeExecutor {
+
+        let contract_input = ContractInput {
+            offer_collateral: margin_coordinator.to_sat() + collateral_reserve_coordinator.to_sat(),
+            accept_collateral: margin_trader.to_sat() + collateral_reserve_trader.to_sat(),
+            fee_rate,
+            contract_infos: vec![ContractInputInfo {
+                contract_descriptor,
+                oracles: OracleInput {
+                    public_keys: vec![to_xonly_pk_29(trade_params.filled_with.oracle_pk)],
+                    event_id,
+                    threshold: 1,
+                },
+            }],
+        };
+
+        let protocol_id = ProtocolId::new();
+        let channel = self.node.inner.get_dlc_channel_by_id(&dlc_channel_id)?;
+        let previous_id = match channel.get_reference_id() {
+            Some(reference_id) => Some(ProtocolId::try_from(reference_id)?),
+            None => None,
+        };
+
+        let temporary_contract_id = self
+            .node
+            .inner
+            .propose_dlc_channel_update(&dlc_channel_id, contract_input, protocol_id.into())
+            .await
+            .context("Could not propose DLC channel update")?;
+
+        let protocol_executor = dlc_protocol::DlcProtocolExecutor::new(self.node.pool.clone());
+        protocol_executor.start_dlc_protocol(
+            protocol_id,
+            previous_id,
+            &temporary_contract_id,
+            &channel.get_id(),
+            DlcProtocolType::Renew {
+                trade_params: (protocol_id, trade_params).into(),
+            },
+        )?;
+
+        db::positions::Position::set_position_to_resizing(
+            conn,
+            peer_id,
+            temporary_contract_id,
+            contracts,
+            coordinator_direction.opposite(),
+            margin_trader,
+            margin_coordinator,
+            average_execution_price,
+            expiry_timestamp,
+            coordinator_liquidation_price,
+            trader_liquidation_price,
+            realized_pnl,
+            order_matching_fee,
+        )?;
+
+        // We add the trade before it is complete so that we can insert the PNL now. Otherwise the
+        // information is lost.
+        db::trades::insert(
+            conn,
+            NewTrade {
+                position_id: position.id,
+                contract_symbol: position.contract_symbol,
+                trader_pubkey: position.trader,
+                quantity: trade_params.quantity,
+                trader_leverage: position.trader_leverage,
+                // TODO: To be removed.
+                coordinator_margin: 0,
+                trader_direction: trade_params.direction,
+                average_price: trade_params
+                    .average_execution_price()
+                    .to_f32()
+                    .expect("to fit"),
+                order_matching_fee,
+                trader_realized_pnl_sat: realized_pnl.map(SignedAmount::to_sat),
+                is_complete: false,
```

Comment:
🤔 It looks to me the only reason why you'd have to create the trade already here is because of the `trader_realized_pnl_sat`. Why not simply add that to the `trade_params` table which is anyways just temporary and intended to remember the state from when the protocol starts to when it ends.

I dislike the `is_complete` flag because

1. It introduces a state on the trade.
2. It is not consistent with the rest of the implementation. e.g. if we want to keep the `is_complete` flag there is no reason for a technical `trade_params` table. We could remove it and create the trade immediately with the required data and set it to "incomplete"

I am not saying we have to stay with the `trade_params` but mixing the two concepts feels wrong to me.

Reply:
You might be right. 

It seemed wrong to use the `trade_params` table because the PNL is not a trade parameter. It's the result of applying the trade to a position.

I agree that it is inconsistent. I can try to fix it, but the priority for me is fairly low.

Reply:
Did it anyway: https://github.com/get10101/10101/pull/2394.

Reply:
Thanks